recipes/isa-l/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+
+import os
+
+required_conan_version = ">=1.33.0"
+
+class LibisalConan(ConanFile):
+    name = "isa-l"
+    description = "Intel's Intelligent Storage Acceleration Library"
+    license = "BSD-3"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/01org/isa-l"
+    topics = ("isa-l", "compression")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC" :True
+    }
+    build_requires = (
+        "libtool/2.4.6",
+        "nasm/2.15.05",
+    )
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.arch not in [ "x86", "x86_64", "armv8" ]:
+            raise ConanInvalidConfiguration("CPU Architecture not supported")
+        if self.settings.os in ["Macos", "Windows"]:
+            raise ConanInvalidConfiguration("Macos and Windows builds not supported")
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            del self.options.with_thread
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            self.run("./autogen.sh")
+            tools.mkdir("_build")
+            with tools.chdir("_build"):
+                env_build = AutoToolsBuildEnvironment(self)
+                extra_args = list()
+                if self.options.shared:
+                    extra_args.extend(('--enable-static=no',))
+                else:
+                    extra_args.extend(('--enable-shared=no',))
+                env_build.configure("../", args=extra_args, build=False, host=False, target=False)
+                env_build.make()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("*/isa-l.h", dst="include/isa-l", keep_path=False)
```

Comment:
we typically dont leave path separators and use `os.path.join` instead

Reply:
I noticed https://github.com/intel/isa-l/blob/62519d97ec8242dce393a1f81593f4f67da3ac92/examples/ec/ec_simple_example.c#L34

is there a reason you are not follow the expectation of the upstream project, it does not look like it should be in a subdir

Reply:
The generated top level include would indicate different:
```
#ifndef _ISAL_H_
#define _ISAL_H_

#define ISAL_MAJOR_VERSION 2
#define ISAL_MINOR_VERSION 30
#define ISAL_PATCH_VERSION 0
#define ISAL_MAKE_VERSION(maj, min, patch)  ((maj) * 0x10000 + (min) * 0x100 + (patch))
#define ISAL_VERSION ISAL_MAKE_VERSION(ISAL_MAJOR_VERSION, ISAL_MINOR_VERSION, ISAL_PATCH_VERSION)

#include <isa-l/crc.h>
#include <isa-l/crc64.h>
#include <isa-l/erasure_code.h>
#include <isa-l/gf_vect_mul.h>
#include <isa-l/igzip_lib.h>
#include <isa-l/mem_routines.h>
#include <isa-l/raid.h>
#endif //_ISAL_H_
```

Reply:
Thank you for filling me in. That makes sense!